offer-manager-domain/src/test/groovy/life/qbic/business/offers/OfferContentSpec.groovy

```diff
@@ -0,0 +1,228 @@
+                .overheadsPMandDS(33)
+                .overheadsDataGeneration(444)
+                .overheadsDataAnalysis(33)
+                .overheadTotal(333)
+                .dataManagementItems(items)
+                .dataAnalysisItems(items)
+                .dataGenerationItems(items)
+                .build()
+
+        OfferContent two = new OfferContent.Builder(customerWithAllAffiliations,externalAcademicAffiliation,projectManager,
+                "2021-10-11","2022-10-11","title","description",
+                "experimental design","O_greiner_ksma_1")
+                .totalVat(3333)
+                .netCost(222)
+                .totalCost(2222)
+                .netDataAnalysis(111)
+                .netDataGeneration(222)
+                .netProjectManagementAndDataStorage(0)
+                .overheadsPMandDS(33)
+                .overheadsDataGeneration(444)
+                .overheadsDataAnalysis(33)
+                .overheadTotal(333)
+                .dataManagementItems(items)
+                .dataAnalysisItems(items)
+                .dataGenerationItems([])
+                .build()
+
+        then: "both are equal"
+        one != two
+
+        one.customerFirstName == two.customerFirstName
+        one.customerLastName == two.customerLastName
+        one.customerTitle == two.customerTitle
+
+        one.customerOrganisation == two.customerOrganisation
+        one.customerStreet == two.customerStreet
+        one.customerPostalCode == two.customerPostalCode
+        one.customerCity == two.customerCity
+        one.customerCountry == two.customerCountry
+        /*Projectmanager*/
+        one.projectManagerFirstName == two.projectManagerFirstName
+        one.projectManagerLastName == two.projectManagerLastName
+        one.projectManagerTitle == two.projectManagerTitle
+        one.projectManagerEmail == two.projectManagerEmail
+
+        /*Project Information*/
+        one.creationDate == two.creationDate
+        one.expirationDate == two.expirationDate
+        one.projectTitle == two.projectTitle
+        one.projectObjective == two.projectObjective
+        one.experimentalDesign == two.experimentalDesign
+        one.offerIdentifier == two.offerIdentifier
+
+        /*Items*/
+        one.dataGenerationItems != two.dataGenerationItems
+        one.dataAnalysisItems == two.dataAnalysisItems
+        one.dataManagementItems == two.dataManagementItems
+
+        /*Overheads*/
+        one.overheadTotal == two.overheadTotal
+        one.overheadsDataGeneration == two.overheadsDataGeneration
+        one.overheadsDataAnalysis == two.overheadsDataGeneration
+        one.overheadsPMandDS == two.overheadsPMandDS
+
+        /*Prices*/
+        one.netDataGeneration == two.netDataGeneration
+        one.netDataAnalysis == two.netDataAnalysis
+        one.netPMandDS != two.netPMandDS
+        one.totalCost != two.totalCost
+        one.netCost == two.netCost
+        one.totalVat == two.totalVat
```

Comment:
This is not what I meant. I was aiming for data driven testing. The question I want this test to answer is: "Are offer contents different if at least one property differs" That means testing all possible combinations of different properties (with at least one being different)

This avoids having a case where one property is unchecked in the equals method.

Reply:
compare: https://github.com/qbicsoftware/offer-manager-2-portlet/pull/728/files#diff-d51b65064c6b02db97f0d94e9640a046eccd0620eb94aee923a31a583c84f682R29

Reply:
I dont know what you mean, can you make a suggestion?

Reply:
I think this is a little overkill, we already have so many fields. Just duplicating all of them looks like this.. not really necessary imo. 
![Screenshot 2021-07-29 at 13 45 02](https://user-images.githubusercontent.com/20778777/127486355-42c1b04e-9f0f-474d-a9a4-87a196ed1d1a.png)